### PR TITLE
added check for metrics' name suffix similar to the one found in prometheus-metrics-core

### PR DESCRIPTION
**Pull request by dpevunov-cp**

### CODE REVIEW

File: core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala [hunk 1]
```diff
@@ -5,8 +5,9 @@ import cats.implicits._
 
 // Mixed into Scala version-specific companion objects
 trait NameCommons {
-  implicit val nameInstances: Show[Name] with Semigroup[Name] with Eq[Name] with Order[Name] =
-    new Show[Name] with Semigroup[Name] with Eq[Name] with Order[Name]{
+  implicit val nameInstances
```
Comment (sideeffffect) [inline core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala:8]: would it be possible to remove this whitespace change?

File: core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala [hunk 2]
```diff
@@ -48,11 +68,12 @@ trait NameCommons {
 
     def impl(s: String): Either[IllegalArgumentException, Suffix] = s match {
       case sufreg(string) => Either.right(new Suffix(string))
-      case _ => Either.left(
-        new IllegalArgumentException(
-          s"Input String - $s does not match regex - ([a-zA-Z0-9_:]*)"
+      case _ =>
```
Comment (sideeffffect) [inline core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala:71]: would it be possible to remove this whitespace change?

**Review by sideeffffect — CHANGES_REQUESTED** [review]
I'd love to minimize the changes as possible. But otherwise the code looks fine, thank you so much for this contribution :heart:

**Review by sideeffffect — APPROVED** [review]
Awesome, let's get this merged and published :pray:

File: core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala (hunk 1 above)
Comment (dpevunov-cp) [reply core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala]: fixed

File: core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala (hunk 2 above)
Comment (dpevunov-cp) [reply core/src/main/scala/io/chrisdavenport/epimetheus/NameCommons.scala]: fixed

**Review by sideeffffect — APPROVED** [review]

**Comment by sideeffffect** [comment]
@dpevunov-cp :heart:

**Pull request merged**